app/src/main/java/org/mozilla/fenix/home/sessioncontrol/viewholders/topsites/TopSitesPagerAdapter.kt

```diff
@@ -33,15 +33,26 @@ class TopSitesPagerAdapter(
         if (payloads.isNullOrEmpty()) {
             onBindViewHolder(holder, position)
         } else {
-            if (payloads[0] is AdapterItem.TopSitePagerPayload) {
+            if (payloads[0] is TopSitePagerPayload) {
                 val adapter = holder.itemView.top_sites_list.adapter as TopSitesAdapter
-                val payload = payloads[0] as AdapterItem.TopSitePagerPayload
-                for (item in payload.changed) {
-                    adapter.notifyItemChanged(
-                        item.first % TOP_SITES_PER_PAGE,
-                        TopSitesAdapter.TopSitePayload(item.second.title)
-                    )
-                }
+                val payload = payloads[0] as TopSitePagerPayload
+
+                update(payload, position, adapter)
+            }
+        }
+    }
+
+    private fun update(
+        payload: TopSitePagerPayload,
+        position: Int,
+        adapter: TopSitesAdapter
+    ) {
+        // Only currently selected page items need to be updated.
+        for (item in payload.changed) {
```

Comment:
Is it possible to destructure item so that we can easily tell that `first` is the page and `second` is the top sites?

Reply:
I'll add a comment explaining this in a kdoc for the method. 👍